hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReaderWrapper.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.common.table.log;
+
+import com.uber.hoodie.common.model.HoodieLogFile;
+import com.uber.hoodie.common.table.log.block.HoodieAvroDataBlock;
+import com.uber.hoodie.common.table.log.block.HoodieCommandBlock;
+import com.uber.hoodie.common.table.log.block.HoodieDeleteBlock;
+import com.uber.hoodie.common.table.log.block.HoodieLogBlock;
+import com.uber.hoodie.exception.HoodieIOException;
+import com.uber.hoodie.exception.HoodieNotSupportedException;
+import org.apache.avro.Schema;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Deque;
+import java.util.List;
+
+public class HoodieLogFormatReaderWrapper implements HoodieLogFormat.Reader {
+
+  // Store the last instant log blocks (needed to implement rollback)
+  private final List<HoodieLogFile> logFiles;
+  private HoodieLogFormatReader currentReader;
+  private final FileSystem fs;
+  private final Schema readerSchema;
+  private final boolean ioIntensiveReader;
+
+  HoodieLogFormatReaderWrapper(FileSystem fs, List<HoodieLogFile> logFiles,
+                                  Schema readerSchema, boolean ioIntensiveReader) throws IOException {
+    this.logFiles = logFiles;
```

Comment:
If you notice `HoodieLogFormatWriter` it handles multiple files and rolling over etc.. I think we should just move the logic here to read across log files inside `HoodieLogFormatReader` as well.

Reply:
hmm, we could do that, merge the wrapper and the reader..